### PR TITLE
[Feature] Support error rate for routine load

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

For routine load, the value is max_error_number (The upper limit of the number of error data lines allowed in the error detection window. When the number of error rows exceeds this value, the import job stops and you need to perform a SHOW ROUTINE LOAD, checking the message in Kafka and correcting the error, based on the ErrorLogUrls. The default value is 0, which means that error lines are not allowed. The error row does not include data filtered out by the WHERE clause.)

For stream load, however, the max_filter_ratio is to specify the maximum fault tolerance of the import job, that is, the maximum proportion of data rows filtered out due to unqualified data quality that the import job can tolerate. Value range: 0 to 1. Default value: 0.

The two import models have different setting for error data tolerance. This PR allows routine load to support fault tolerance in the same way that stream load does. You can specify the max_filter_ratio property when creating an routine load task:
```
CREATE ROUTINE LOAD dbyzr.aaa ON clickbench
COLUMNS(...)
PROPERTIES
(
    "desired_concurrent_number"="5",
    "format" ="json",
    "max_filter_ratio" = "0.3" 
)
FROM KAFKA
(
	...
);
```
When importing a batch of data, the task will be paused if the proportion of incorrect data exceeds max_filter_ratio.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/runtime/routine_load/routine_load_task_executor.cpp [hunk 1]
```diff
@@ -277,7 +277,11 @@ Status RoutineLoadTaskExecutor::submit_task(const TRoutineLoadTask& task) {
     // the routine load task'txn has alreay began in FE.
     // so it need to rollback if encounter error.
     ctx->need_rollback = true;
-    ctx->max_filter_ratio = 1.0;
+    if (task.__isset.max_filter_ratio) {
+        ctx->max_filter_ratio = task.max_filter_ratio;
```
[the remark this answers is not in the document]
Comment (zaorangyang) [reply be/src/runtime/routine_load/routine_load_task_executor.cpp:281]: > How to check if the `error rows / total rows` is larger than max_filter_ratio?

[the remark this answers is not in the document]
Comment (zaorangyang) [reply be/src/runtime/routine_load/routine_load_task_executor.cpp:281]: image

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java [hunk 2]
```diff
@@ -298,6 +299,10 @@ protected void setOptional(CreateRoutineLoadStmt stmt) throws UserException {
         if (stmt.getMaxErrorNum() != -1) {
             this.maxErrorNum = stmt.getMaxErrorNum();
         }
+        if (Math.abs(stmt.getMaxFilterRatio() - 1) > 0.001) {
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java:302]: 0.9999999 will not be set?

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java (hunk 2 above)
Comment (zaorangyang) [reply fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java:300]: > 0.9999999 will not be set?

updated

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 37 (62.16%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/PulsarTaskInfo.java|0|2|00.00%|[155, 156]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaTaskInfo.java|0|2|00.00%|[185, 186]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|6|12|50.00%|[538, 539, 1061, 1062, 1063, 1670]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterRoutineLoadStmt.java|7|10|70.00%|[158, 159, 162]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|7|8|87.50%|[507]|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|3|3|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by zaorangyang** [comment]
@Mergifyio backport branch-3.0

**Comment by zaorangyang** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**